components/script/layout_wrapper.rs

```diff
@@ -757,46 +779,48 @@ impl<'ln> ServoThreadSafeLayoutNode<'ln> {
     }
 }
 
+// NB: The implementation here is a bit tricky because elements implementing
+// pseudos are supposed to return false for is_element().
 impl<'ln> NodeInfo for ServoThreadSafeLayoutNode<'ln> {
     fn is_element(&self) -> bool {
         self.pseudo == PseudoElementType::Normal && self.node.is_element()
     }
 
     fn is_text_node(&self) -> bool {
-        // It's unlikely that text nodes will ever be used to implement a
-        // pseudo-element, but the type system doesn't really enforce that,
-        // so we check to be safe.
-        self.pseudo == PseudoElementType::Normal && self.node.is_text_node()
+        self.node.is_text_node()
     }
 
     fn needs_layout(&self) -> bool {
-        self.pseudo != PseudoElementType::Normal ||
-        self.node.is_element() || self.node.is_text_node()
+        self.node.is_text_node() || self.node.is_element()
```

Comment:
We still can. The iterator returned by children() operates on nodes of type Self, and then filters them using needs_layout before returning them. So if there's a comment node, we'll still create a ServoThreadSafeLayoutNode for it, just not return it to the traversal code.

So I think we need this.